api/src/main/java/io/cloudevents/CloudEvent.java

```diff
@@ -33,45 +33,60 @@
 @ParametersAreNonnullByDefault
 public interface CloudEvent {
 
-    /**
-     * The event context attributes
-     */
-    Attributes getAttributes();
```

Comment:
If `CloudEvent` implements `Attributes`, then this method can still live here.

The reason to keep this method is to allow forward compatibility or backward compatibility with old/new attributes, accessing some specific "features" of a specific spec version

Reply:
Isn't casting to `CloudEventsV1Impl` would serve the same purpose?

Reply:
User of SDK can still downcast it to `CloudEventImpl` and we can certainly even make it public there for reason you just explained, but with regard to `CloudEvent` interface we should start looking at it as something that should never change.

Reply:
In go we had been casting to a version to access but it turns out that putting the accessors at the top level like you might in java style code and internally redirect to the correct attribute was pretty valuable for moving code forward in the cloudevent versions.

We choose to keep the accessor interface to be aligned with the most current version of the spec and swizzle internally to make older encodings look like new ones.

Reply:
@n3wscott I can get behind it and as you can see at the moment that is exactly what we're doiing